docs/docs/development/studio-setup.md

```diff
@@ -0,0 +1,96 @@
+---
+title: Studio Setup
+---
+
+:::warning[Alpha Feature]
+
+ZMK Studio support is in alpha. Although best efforts are being made, backwards compatibility during active development is not guaranteed.
+
+:::
+
+This guide will walk you through enabling ZMK Studio support for a keyboard.
+
+The main additional pieces needed for Studio support involve additional metadata needed in order
+to properly to display the physical layouts available for the particular keyboard.
+
+# Physical Layout Positions
+
+Physical layouts are described as part of the [new shield guide](./new-shield.mdx#physical-layouts) with the exception of the `keys` property that is required for Studio support. This is used to describe the physical attributes of each key position present in that layout and its items are listed in the same order as keymap bindings, matrix transforms, etc. The properties available are:
+
+| Property   | Type     | Description                          | Unit                                                    |
+| ---------- | -------- | ------------------------------------ | ------------------------------------------------------- |
+| Width      | int (>0) | Key(cap) width                       | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+| Height     | int (>0) | Key(cap) height                      | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+| X          | uint     | Key X position (top-left point)      | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+| Y          | uint     | Key Y position (top-left point)      | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+| Rotation   | int      | Key rotation (positive => clockwise) | [centi-](https://en.wikipedia.org/wiki/Centi-)degree    |
+| Rotation X | int      | Rotation origin X position           | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+| Rotation Y | int      | Rotation origin Y position           | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
```

Comment:
```suggestion
| Rotation   | int      | Key rotation (positive &rArr; clockwise) | [centi-](https://en.wikipedia.org/wiki/Centi-)degree    |
| Rotation X | int      | Rotation origin relative X position           | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
| Rotation Y | int      | Rotation origin relative Y position           | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
```

Reply:
Really splitting up the conversation here, but I wanted to grab the rArr for you :sweat_smile: What happens when you rotate around 0,0 - I thought it would rotate around the key center, but if it rotates around the 0,0 of the keyboard then that's fine as-is

Reply:
Yeah, it's an absolute rotation point, which is useful for a few scenarios, e.g. rotating a whole half of a layout by assigning the same rotation, and rotation X/Y to all of the relevant key positions..